core/templates/pages/exploration-editor-page/history-tab/history-tab.component.ts

```diff
@@ -71,7 +71,12 @@ angular.module('oppia').component('historyTab', {
       // Compares the two selected versions and displays the comparison
       // results.
       ctrl.compareSelectedVersions = function() {
-        if (ctrl.selectedVersionsArray.length === 2) {
+        // The selectedVersionsArray may contain empty elements as the
+        // array items are directly manipulated based on indices on change.
+        if (
+          Number.isInteger(ctrl.selectedVersionsArray[0]) &&
+          Number.isInteger(ctrl.selectedVersionsArray[1])
```

Comment:
Just to be safe, should we also keep the length check on `selectedVersionsArray`?

Reply:
Done